docker-compose-app.yml

```diff
@@ -6,6 +6,26 @@ services:
     build: web
     ports:
       - 80:3000
+    networks:
+      - default
+    deploy:
+      replicas: 3
+      update_config:
+        parallelism: 1
+        delay: 5s
+      restart_policy:
+        condition: on-failure
+      placement:
+        constraints:
+          - node.role == worker
```

Comment:
Yes, it would schedule across all nodes. Not really a problem in a small local swarm. I guess in the cloud you'd have an ASG set to 3 or 5 nodes for managers and one with min and max scaling for workers, so it would make sense to constrain to workers in this case. Some services (like the visualiser) need to run on manager nodes (I also run the private registry on a manager to save it moving around as it has state)